### PR TITLE
Added features

### PR DESCRIPTION
**Pull request by dhruv282**
- The user gets an option for a particular screenshot to be retaken in the summary page from the image grid.

- New cache system implemented. This new system sends date range information from the cache instead of fetching the timemap again from the archive. Also, if the user requests a date range for a URI that is not cached, after responding to that request, a silent process of calculating the simhash of the full timemap is executed then stored to the cache.

- Zoomable histogram added to the histogram page.

- Added support to handle redirected pages from archives.

- Updated ReadMe file to reflect new changes.

- Added appropriate error messages for date range input.

### CODE REVIEW

File: public/index.html [hunk 1]
```diff
@@ -17,14 +17,15 @@
...
+		<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">
 		<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js"></script>
 		<script src="//d3js.org/d3.v4.min.js"></script>
 		<script src="https://code.jquery.com/jquery-migrate-3.1.0.js"></script>
 
     	<script src="/scripts/imageSliderScript.js"></script>
     	<script src="/scripts/imageGridScript.js"></script>
     	<script src="/scripts/timeline.js?version=1005"></script>
-	<script src="/scripts/histogram.js"></script>
+		<script src="/scripts/histogram.js"></script>
```
Comment (ibnesayeed) [inline public/index.html]:
```suggestion
	<script src="/scripts/histogram.js"></script>
```

This is an unnecessary change.

**Comment by ibnesayeed** [comment]
Let me know when it is ready to be merged.

**Comment by dhruv282** [comment]
It is ready to be merged, thanks!
